### PR TITLE
Move performance to java agent, merge javadoc into API page

### LINKED ISSUE #5211: Finish refactoring Java information architecture
Followup on #4966 to finish the ideas in #4853. If / when complete, we can decide if its useful for other languages to follow suit.

- [] Refactor [Instrumentation page](https://opentelemetry.io/docs/languages/java/instrumentation/) into pages discussing how to use the API, guide for library developers, context propagation, using semantic conventions
- [] Nest [Registry](https://opentelemetry.io/docs/languages/java/registry/), [Performance](https://opentelemetry.io/docs/languages/java/performance/), [API (Javadoc)](https://opentelemetry.io/docs/languages/java/api/) under new "Learn more" section
- [] Add new Learn the Basics page, introducing users to the otel java ecosystem and routing them to various pages for more information
- [] Add page describing instrumentation landscape in java and guide to installing instrumentation in applications

### PR DESCRIPTION
**Pull request by jack-berg**
Resolves #5211.

An alternative to #5522 based on the [conversation here](https://github.com/open-telemetry/opentelemetry.io/pull/5522#issuecomment-2460585697):

- Move "Performance" page to "Zero-code Instrumentation -> Java -> Agent -> Performance"
- Merge "Javadoc" into "Record Telemetry with API", page from "../api-components/" to "../api/". 

---

### Previews

- https://deploy-preview-5590--opentelemetry.netlify.app/docs/languages/java/api/
- https://deploy-preview-5590--opentelemetry.netlify.app/docs/zero-code/java/agent/performance/

### CODE REVIEW

File: content/en/docs/zero-code/java/agent/performance.md
Comment (chalin) [inline content/en/docs/zero-code/java/agent/performance.md:1]: This page will need an alias.

Comment (chalin) [reply content/en/docs/zero-code/java/agent/performance.md:1]: And an adjusted weight so that it is probably last (at least after the config pages) -- WDTY?

File: content/en/docs/languages/java/api.md
Comment (chalin) [inline content/en/docs/languages/java/api.md:58]:
```suggestion
components, see [javadoc.io/doc/io.opentelemetry(https://javadoc.io/doc/io.opentelemetry).
```

File: content/en/docs/languages/java/api.md
Comment (chalin) [inline content/en/docs/languages/java/api.md:57]: How about just "the" instead of "an exhuastive"?

```suggestion
{{% alert title=Javadoc %}} For the Javadoc reference of all OpenTelemetry Java
```

**Review by jaydeluca — APPROVED** [review]

**Review by chalin — APPROVED** [review]
Works for me. @open-telemetry/docs-approvers any other feedback?

**Review by theletterf — APPROVED** [review]

**Review by chalin — CHANGES_REQUESTED** [review]
Blocking until I figure out how best to manage the broken links in non-en pages.

**Comment by chalin** [comment]
@jack-berg - there are broken links in this `en` page that need to be fixed:

```
docs/zero-code/java/agent/performance/index.html
  target does not exist --- docs/zero-code/java/agent/performance/index.html --> ../sdk/#sampler
  hash does not exist --- docs/zero-code/java/agent/performance/index.html --> ../getting-started/#prerequisites
```

**Comment by chalin** [comment]
@jack-berg - more broken links:

```
docs/zero-code/java/agent/performance/index.html
  target does not exist --- docs/zero-code/java/agent/performance/index.html --> ../sdk/#sampler
  hash does not exist --- docs/zero-code/java/agent/performance/index.html --> ../getting-started/#prerequisites
docs/collector/deployment/agent/index.html
  target does not exist --- docs/collector/deployment/agent/index.html --> /docs/languages/java/api-components/#meterprovider
docs/languages/java/intro/index.html
  target does not exist --- docs/languages/java/intro/index.html --> ../performance/
```

**Comment by jack-berg** [comment]
> Blocking until I figure out how best to manage the broken links in non-en pages.

I've fixed all the links and believe there were'nt any links in non-en pages

**Review by svrnm — APPROVED** [review]
LGTM, will wait for @chalin to re-review.

**Review by chalin — APPROVED** [review]
It seems like the non-en link-check failures were in en fallback pages.

LGTM

**Comment by chalin** [comment]
It seems that we need one more approval from @open-telemetry/java-approvers.

**Comment by jack-berg** [comment]
> It seems that we need one more approval from @open-telemetry/java-approvers.

We've got an approval from java-approver @jaydeluca. Do we need a second? In the past, one java-approver approval has been enough. Not expressing an opinion on whether we should have one or two approvals required - just clarifying my understanding. 🙂

**Comment by chalin** [comment]
I'm not sure what the policy is; I'll let @svrnm chime in. Also, I don't see @jaydeluca listed in @open-telemetry/java-approvers, which matches what GH reported above:

> image

**Comment by jack-berg** [comment]
> Also, I don't see @jaydeluca listed in @open-telemetry/java-approvers, which matches what GH reported above:

Ah that's right. Yeah we have a bit of a weird thing going on where we have two partially independent / partially overlapping groups of approvers in the java SIG: @open-telemetry/java-approvers and @open-telemetry/java-instrumentation-approvers. Both groups are trusted collaborators in the Java SIG, but the @open-telemetry/java-instrumentation-approvers tend to contribute more to the `opentelemetry-java-instrumentation` repo while the @open-telemetry/java-approvers tend to contribute more to the `opentelemetry-java` repo.

In my head, the set of people trusted to approve java opentelemetry.io PRs is the superset of both groups, since the docs have much less strict delineation between content related to `opentelemetry-java-instrumentation` and `opentelemetry-java`.

Some instances where an approval from a @open-telemetry/java-instrumentation-approvers has been sufficient: #5471, #5275, #4966

**Comment by chalin** [comment]
Thanks for the explanation @jack-berg. That works for me. Adding this to the merge queue.

**Comment by trask** [comment]
> In my head, the set of people trusted to approve java opentelemetry.io PRs is the superset of both groups

👍

**Pull request merged**
